portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +34,69 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private final static Logger LOGGER = Logger.getLogger(DataServlet.class.getName());
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Get all the comments from the database
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comment");
+      comments.add(comment);
+    }
+
+    response.setContentType("application/json");
+    response.getWriter().println(toJSONString(comments, "comments"));
+  }
+
+  /*
+   * Helper function to store an array in JSON format
+   */
+  private String toJSONString(List<String> array, String arrayName) {
+    Gson gson = new Gson();
+    List<String> jsonArray = new ArrayList<String>();
+
+    if (array != null) {
+      for (String string : array) {
+        jsonArray.add(gson.toJson(string));
+      }
+      String json = "{ \"" + arrayName + "\" :"
```

Comment:
Can gson be used to convert the entire collection instead of a single element at a time?

Reply:
Yep done now